lib/http-client.ts

```diff
@@ -54,7 +55,10 @@ export class HttpClient implements IRangeRequestClient {
   }
 
   public getHeadInfo(): Promise<IHeadRequestInfo> {
-    return _fetch(this.url, {method: 'HEAD'}).then(resp => HttpClient.makeResponse(resp));
+    return _fetch(this.url, {method: 'HEAD'}).then(resp => {
+      this.url = resp.url;
```

Comment:
Assigning the URL from the response back to the request, looks a bit odd.

Reply:
What do you mean by request? Isn't `this` the instance of the `HttpClient`? 🤔 The goal here is to change the `httpClientInstance.url`, or in other words the url used on line 58 as the first argument to `_fetch`.

Reply:
HttpClient maybe is a confusing name. The HttpClient is the request. It is bound to one, and only URL request.

Reply:
Maybe create a second url attribute for fetching the data?

So fetch does already handle the redirect right? That is why you use the url?

Reply:
> So fetch does already handle the redirect right? That is why you use the url?

Yes.